app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentList.kt

```diff
@@ -55,6 +72,44 @@ fun DocumentListScreen(
   val documents = documentViewModel.documents.collectAsState()
   documentViewModel.getDocuments()
 
+  val functions = FirebaseFunctions.getInstance("europe-west9")
+  val context = LocalContext.current
+  val scannerOptions =
+      GmsDocumentScannerOptions.Builder()
+          .setScannerMode(SCANNER_MODE_FULL)
+          .setGalleryImportAllowed(true)
+          .setResultFormats(GmsDocumentScannerOptions.RESULT_FORMAT_PDF)
+          .build()
+  val scanner = GmsDocumentScanning.getClient(scannerOptions)
+  val scannerLauncher =
+      rememberLauncherForActivityResult(
+          contract = ActivityResultContracts.StartIntentSenderForResult()) {
+            if (it.resultCode == Activity.RESULT_OK) {
+              val scanningResult = GmsDocumentScanningResult.fromActivityResultIntent(it.data)
+              Log.d("DocumentList", "Scanning result: $scanningResult")
+              scanningResult?.pdf?.let { pdf ->
+                val bytes = pdf.uri.toFile().readBytes()
+                val bytes64 = Base64Utils.encodeUrlSafe(bytes)
+                val scanTimestamp = Timestamp.now().seconds
+                functions
+                    .getHttpsCallable("storeDocument")
+                    .call(
+                        mapOf(
+                            "content" to bytes64,
+                            "fileFormat" to "application/pdf",
+                            "title" to "Scan ${scanTimestamp}.pdf",
+                            "travelId" to "$scanTimestamp",
```

Comment:
I updated the other fields according to the two types of file formats generated by the scanner. For the travel ID it will be set in the PR handling travels isolation

Reply:
Ok it has been implemented anyway